internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -38,12 +41,39 @@ func (f *FirstReadyCluster) FindCluster(kafka *dbapi.KafkaRequest) (*api.Cluster
 		SupportedInstanceType: kafka.InstanceType,
 	}
 
-	cluster, err := f.ClusterService.FindCluster(criteria)
-	if err != nil {
-		return nil, err
+	clusters, findAllClusterErr := f.ClusterService.FindAllClusters(criteria)
+	if findAllClusterErr != nil {
+		return nil, findAllClusterErr
+	}
+
+	// Get total number of streaming unit used per region and instance type
+	streamingUnitCountPerRegionList, countStreamingUnitErr := f.KafkaService.CountStreamingUnitByRegionAndInstanceType()
+	if countStreamingUnitErr != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, countStreamingUnitErr, "failed to get count of streaming units by region and instance type")
+	}
+
+	instanceSize, getInstanceSizeErr := f.KafkaConfig.GetKafkaInstanceSize(kafka.InstanceType, kafka.SizeId)
+	if getInstanceSizeErr != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, getInstanceSizeErr, "failed to get instance size for kafka '%s'", kafka.ID)
+	}
+
+	// Find first ready cluster that has remaining capacity (total streaming unit used by existing and requested kafka is less than the cluster maxUnit)
+	for _, cluster := range clusters {
+		currentStreamingUnitsUsed := streamingUnitCountPerRegionList.GetStreamingUnitCountForClusterAndInstanceType(cluster.ClusterID, kafka.InstanceType)
+
+		capacityInfo, getCapacityInfoErr := cluster.RetrieveDynamicCapacityInfo()
+		if getCapacityInfoErr != nil {
+			return nil, errors.NewWithCause(errors.ErrorGeneral, getCapacityInfoErr, "failed to retrieve capacity information for cluster %s", cluster.ClusterID)
+		}
+		maxStreamingUnits := capacityInfo[kafka.InstanceType].MaxUnits
+
+		if currentStreamingUnitsUsed+instanceSize.CapacityConsumed < int(maxStreamingUnits) {
```

Comment:
```suggestion
		if currentStreamingUnitsUsed+instanceSize.CapacityConsumed <= int(maxStreamingUnits) {
```
I think, this needs to be `<=`. e.g if maxUnit is `2`, capacityUsed is `1` and `capacityConsumed` is `1` as well, we want to accept the kafka as it can be placed onto the cluster. With `<` we'll end up in a situation where we do not accept this kafka onto this cluster.